AK/Platform.h

```diff
@@ -249,6 +249,13 @@
 #    define NO_UNIQUE_ADDRESS [[no_unique_address]]
 #endif
 
+#if defined(AK_OS_WINDOWS)
+// https://learn.microsoft.com/en-us/cpp/cpp/empty-bases?view=msvc-170
+#    define WINDOWS_ENABLE_EBCO __declspec(empty_bases)
+#else
+#    define WINDOWS_ENABLE_EBCO
```

Comment:
`ENABLE_EMPTY_BASE_CLASS_OPTIMIZATION` ? 

EBCO doesn't tell you anything without going to the definition of the macro and then to the link.

`AK_COMPACT_EMPTY_BASES` ? We should apply that to other classes that expect EBCO as well. Likely only some generics in AK. Tuple, Pair, TypeList, ...